python-frontend/src/test/java/org/sonar/python/tree/IPythonTreeMakerTest.java

```diff
@@ -67,85 +77,78 @@ public void line_magic() {
-    lineMagic = findFirstChildOf(lineMagicStatement, Tree.Kind.LINE_MAGIC);
-    assertThat(lineMagic).isNotNull();
+      "b = a % foo(b)", treeMaker::fileInput);
+    statementList = findFirstChildWithKind(parse, Tree.Kind.STATEMENT_LIST);
+    assertThat(statementList.children()).hasSize(4);
+    assertThat(statementList.children().get(0).getKind()).isEqualTo(Tree.Kind.EXPRESSION_STMT);
+    assertThat(statementList.children().get(1).getKind()).isEqualTo(Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(statementList.children().get(2).getKind()).isEqualTo(Tree.Kind.COMPOUND_ASSIGNMENT);
+    assertThat(statementList.children().get(3).getKind()).isEqualTo(Tree.Kind.ASSIGNMENT_STMT);
+    assertThat(findFirstChildWithKind(statementList.children().get(3), Tree.Kind.MODULO)).isNotNull();
 
     parse = parseIPython("print(b)\n" +
       "%timeit a = foo(b); b = 2\n" +
-      "a += b\n" +
-      "print(a)", treeMaker::fileInput);
-    assertThat(parse).isNotNull();
-    lineMagicStatement = findFirstChildOf(parse, Tree.Kind.LINE_MAGIC_STATEMENT);
-    assertThat(lineMagicStatement)
-      .isNotNull()
-      .isInstanceOf(LineMagicStatement.class);
-
-    assertThat(lineMagicStatement.children()).hasSize(1);
-    lineMagic = findFirstChildOf(lineMagicStatement, Tree.Kind.LINE_MAGIC);
-    assertThat(lineMagic).isNotNull();
-
-    parse = parseIPython("print(b)\n" +
-      "%autocall 1\n", treeMaker::fileInput);
-    assertThat(parse).isNotNull();
+      "a += b", treeMaker::fileInput);
+    statementList = findFirstChildWithKind(parse, Tree.Kind.STATEMENT_LIST);
+    assertThat(statementList.children()).hasSize(3);
+    assertThat(statementList.children().get(0).getKind()).isEqualTo(Tree.Kind.EXPRESSION_STMT);
+    assertThat(statementList.children().get(1).getKind()).isEqualTo(Tree.Kind.LINE_MAGIC_STATEMENT);
+    assertThat(statementList.children().get(2).getKind()).isEqualTo(Tree.Kind.COMPOUND_ASSIGNMENT);
   }
 
   @Test
   public void assignment_rhs_test() {
     var parse = parseIPython("print(b)\n" +
       "a = yield foo(b)\n" +
-      "c = bar(a) + b", treeMaker::fileInput);
+      "c = bar(a) + b\n" +
+      "d = bar(c) % bar(a)", treeMaker::fileInput);
     assertThat(parse).isNotNull();
 
-    var assignments = findChildrenOf(parse, Tree.Kind.ASSIGNMENT_STMT);
-    assertThat(assignments).hasSize(2);
-    var yieldExpression = findFirstChildOf(assignments.get(0), Tree.Kind.YIELD_EXPR);
+    var assignments = findChildrenWithKind(parse, Tree.Kind.ASSIGNMENT_STMT);
+    assertThat(assignments).hasSize(3);
+    var yieldExpression = findChildrenWithKind(assignments.get(0), Tree.Kind.YIELD_EXPR);
     assertThat(yieldExpression).isNotNull();
-    var binaryExpression = findFirstChildOf(assignments.get(1), Tree.Kind.PLUS);
+    var binaryExpression = findChildrenWithKind(assignments.get(1), Tree.Kind.PLUS);
+    assertThat(binaryExpression).isNotNull();
+    binaryExpression = findChildrenWithKind(assignments.get(2), Tree.Kind.MODULO);
```

Comment:
Same here, probably it's better to use the fact that it's an assignment and get the assigned value through `AssignmentStatement#assignedValue`